src/main/java/game/entities/items/weapons/Damage.java

```diff
@@ -1,15 +1,28 @@
 package game.entities.items.weapons;
 
 import game.entities.Entity;
+import game.entities.items.Item;
 
 public class Damage {
     private Entity target;
+    private Weapon weapon;
 
     public Damage(Entity target) {
         this.target = target;
     }
 
-    public void dealDamage(int damageDealt) { target.setHP(target.getHP() - damageDealt); }
+    public Damage(Weapon weapon) {
+        this.weapon = weapon;
+    }
+
+    public Damage addDamage(int damageDealt) { 
+        target.setHP(target.getHP() - damageDealt); 
+        return this;
+    }
+    public Damage addDamagetoWeapon(int damageDealt) {
+        weapon.setDurability(weapon.getDurability() - damageDealt);
```

Comment:
Change this so that there is only one method that adds damages the thing that is receiving the damage (in this case, any object that is a `GameObject`)

Reply:
it takes a Damageable instead of GameObject asshole